R/helpers.R

```diff
@@ -13,6 +13,46 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+.geocode <- function(name, country, method = "osm") {
+  if (missing(name) || !is.character(name)) {
+    stop("Please provide a valid location name as a string.")
+  }
+  
+  # Use match.arg to validate the method parameter
+  method <- match.arg(method, choices = c("osm", "census", "arcgis", "census_simple", "geocodio",
```

Comment:
You don't need this call to `match.arg()` if you do this in the top-level `new_city()` function, but then you need to pass `.geocode(name, country, method)` in the `new_city()` function.

Reply:
you are totally right !